src/resources/views/crud/fields/select2_from_ajax.blade.php

```diff
@@ -58,115 +57,152 @@
+                                    return {
+                                        q: params.term, // search term
+                                        page: params.page, // pagination
+                                    };
                                 }
-                            }),
-                           pagination: {
-                                 more: data.current_page < data.last_page
-                           }
-                        };
-
-                        return result;
-                    },
-                    cache: true
-                },
-            });
-
-            // if any dependencies have been declared
-            // when one of those dependencies changes value
-            // reset the select2 value
-            for (var i=0; i < $dependencies.length; i++) {
-                $dependency = $dependencies[i];
-                $('input[name='+$dependency+'], select[name='+$dependency+'], checkbox[name='+$dependency+'], radio[name='+$dependency+'], textarea[name='+$dependency+']').change(function () {
-                    element.val(null).trigger("change");
-                });
+                            },
+                            processResults: function (data, params) {
+                                params.page = params.page || 1;
+
+                                var result = {
+                                    results: $.map(data.data, function (item) {
+                                        textField = $fieldAttribute;
+                                        return {
+                                            text: item[textField],
+                                            id: item[$connectedEntityKeyName]
+                                        }
+                                    }),
+                                   pagination: {
+                                         more: data.current_page < data.last_page
+                                   }
+                                };
+
+                                return result;
+                            },
+                            cache: true
+                        },
+                    });
+
+                    // if any dependencies have been declared
+                    // when one of those dependencies changes value
+                    // reset the select2 value
+                    for (var i=0; i < $dependencies.length; i++) {
+                        $dependency = $dependencies[i];
+                        $('input[name='+$dependency+'], select[name='+$dependency+'], checkbox[name='+$dependency+'], radio[name='+$dependency+'], textarea[name='+$dependency+']').change(function () {
+                            element.val(null).trigger("change");
+                        });
+                    }
+                }
+                $value = element.attr('data-value');
+                if (typeof $value !== typeof undefined && $value !== false) {
```

Comment:
I think it would be easier to understand and cleaner if we used ```element.attr('data-value')``` directly, instead of defining a ```$value``` variable just for this one usage.

Reply:
I used the variable so I don't have to query the DOM twice checking if undefined and if is false. 

Not sure what the benchmarks for this, it's just the way I think about it. 

If you feel or are sure, querying the DOM twice is "cheaper" than setting up a variable I am ok with it.